master/internal/agent/agents.go

```diff
@@ -38,6 +40,30 @@ func (a *agents) Receive(ctx *actor.Context) error {
 			ctx.Respond(errors.Errorf("agent already connected: %s", id))
 			return nil
 		}
+	case *proto.GetAgentsRequest:
+		response := &proto.GetAgentsResponse{}
+		for _, a := range a.summarize(ctx) {
+			if msg.Label != "" && msg.Label != a.Label {
+				continue
+			}
+			response.Agents = append(response.Agents, toProtoAgent(a))
+		}
+		sort.Slice(response.Agents, func(i, j int) bool {
+			switch a1, a2 := response.Agents[i], response.Agents[j]; msg.SortBy {
+			case proto.GetAgentsRequest_SORT_BY_TIME:
+				return a1.RegisteredTime.Seconds < a2.RegisteredTime.Seconds
+			case proto.GetAgentsRequest_SORT_BY_UNSPECIFIED, proto.GetAgentsRequest_SORT_BY_ID:
+				fallthrough
+			default:
+				return a1.Id < a2.Id
```

Comment:
question: curious why take this approach instead of using `return a1.ID < a2.ID` as part of the SORT_BY_ID block, and a panic here?

The current approach seems a tad dangerous. If we were to add another sort type and someone were to put it below `SORT_BY_ID` it wouldn't work correctly.